.github/workflows/CD.yml

```diff
@@ -1,78 +1,84 @@
 name: CD
 
 on:
+  workflow_dispatch:
   push:
     branches: [master]
 
 env:
   CARGO_TERM_COLOR: always
+
 jobs:
-  docker:
+  build:
     runs-on: ubuntu-latest
     steps:
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
-
       - name: Login to DockerHub
-        uses: docker/login-action@v1
+        uses: docker/login-action@v2.0.0
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Build and push
-        id: docker_build
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v3.1.0
         with:
           file: ../../docker/Dockerfile
           push: true
           tags: ghcr.io/gear-tech/node:latest
 
   deploy:
     runs-on: ubuntu-latest
-    needs: docker
+    needs: build
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - name: Read secrets
+        uses: hashicorp/vault-action@v2.4.1
```

Comment:
Is this vault backed up and available for somebody except you? It should be protected against the bus factor.